### PR TITLE
SDK 945 & 952 - add accessibility for Journey form and autocomplete

### PR DESCRIPTION
**Pull request by nicolas-chaix**
Fixes: SDK-945, SDK-952, SDK-954, SDK-965

### CODE REVIEW

File: navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/form/FormFragment.java [hunk 1]
```diff
@@ -165,6 +176,10 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
     public void onResume() {
         super.onResume();
         defaultUi();
+
+        new Handler().postDelayed(
+                () -> departureLabel.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_FOCUSED),
+                300);
```
Comment (Guitouille) [inline navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/form/FormFragment.java]: I think the 300ms should be put in a constant.

File: navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/form/FormFragment.java [hunk 2]
```diff
@@ -183,34 +198,59 @@ private void setOriginDestinationClickListener(final TextView label, final int l
 
     private void defaultUi() {
         String place;
+        String description;
```
Comment (Guitouille) [inline navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/form/FormFragment.java]: Please add an empty line after.

File: navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/journeys/JourneysFragment.java [hunk 3]
```diff
@@ -75,12 +74,12 @@
 
     @Inject
     JourneysViewModel journeysViewModel;
-    private InternalJourneyRequest internalJourneyRequest;
+    private InternalJourneyRequest internalJourneysRequest;
```
Comment (Guitouille) [inline navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/journeys/JourneysFragment.java]: I disagree with that change as there is only 1 request. Moreover, if we change the variable name, we should also change the class name. Maybe it is just badly named ?

File: navitiasdkux/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -14,4 +14,18 @@
...
+        <item name="android:textColor">@color/selector_form_address</item>
+        <item name="android:textSize">16sp</item>
+        <item name="android:textStyle">bold</item>
+    </style>
+
+    <style name="AddressIconStyle">
+        <item name="android:layout_marginEnd">8dp</item>
+        <item name="android:textSize">24sp</item>
+    </style>
+</resources>
```
Comment (Guitouille) [inline navitiasdkux/src/main/res/values/styles.xml]: Please add an empty line after.

File: sample/src/main/java/org/kisio/sample/navitia/sdk/ui/MainActivity.java [hunk 5]
```diff
@@ -40,8 +40,8 @@ protected void onCreate(Bundle savedInstanceState) {
             final Intent intent = new Intent(v.getContext(), FormActivity.class);
             final JourneysRequest request = new JourneysRequest("fr-idf");
             request
-                    .setOriginId("2.3665844;48.8465337")
-                    .setDestinationId("2.2979169;48.8848719")
+//                    .setOriginId("2.3665844;48.8465337")
```
Comment (Guitouille) [inline sample/src/main/java/org/kisio/sample/navitia/sdk/ui/MainActivity.java]: Remove that commented code ?

File: navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/autocomplete/AutoCompleteFragment.java [hunk 6]
```diff
@@ -198,164 +181,63 @@ public void onProviderEnabled(String provider) {
...
+        criteria.setAltitudeRequired(false);
+        criteria.setSpeedRequired(false);
+        criteria.setBearingRequired(false);
+        criteria.setVerticalAccuracy(Criteria.ACCURACY_LOW);
 
-        if (requestFocus) {
-            editText.requestFocusFromTouch();
-            editText.sendAccessibilityEvent(AccessibilityEvent.TYPE_VIEW_FOCUSED);
-        }
+        locationManager.requestLocationUpdates(20_000,
```
[the remark this answers is not in the document]
Comment (nicolas-chaix) [reply navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/autocomplete/AutoCompleteFragment.java]: converting min time and min distance as constant

File: navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/component/address/ClearableEditText.java [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+                if (!internalJourneyRequest.getJourneysRequest().getOriginId().isEmpty()) {
+                    place = internalJourneyRequest.getJourneysRequest().getOriginLabel().isEmpty()
+                            ? internalJourneyRequest.getJourneysRequest().getOriginId()
+                            : internalJourneyRequest.getJourneysRequest().getOriginLabel();
+                }
+            } else {
+                place = internalJourneyRequest.getOriginAddress();
+            }
+
+            // Arrival
```
[the remark this answers is not in the document]
Comment (nicolas-chaix) [reply navitiasdkux/src/main/java/org/kisio/navitia/sdk/ui/presentation/component/address/ClearableEditText.java]: this is the default indentation from android studio.
I'm moving comments inside the conditions

**Pull request merged**